differ/differ_test.go

```diff
@@ -1451,3 +1453,116 @@ func TestProcessClustersWeeklyDigest(t *testing.T) {
+			TemplateRendererEndpoint: "/render_reports",
+		},
+	}
+	serviceLogEventThresholds.TotalRisk = 1
+	serviceLogEventFilter = "totalRisk > totalRiskThreshold"
+
+	errorKeys := map[string]utypes.RuleErrorKeyContent{
+		"RULE_1": {
+			Metadata: utypes.ErrorKeyMetadata{
+				Description: "This reason is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long.",
+				Impact: utypes.Impact{
+					Name:   "impact_1",
+					Impact: 3,
+				},
+				Likelihood: 2,
+			},
+			Reason:    "rule 1 reason",
+			HasReason: true,
+		},
+		"RULE_2": {
+			Metadata: utypes.ErrorKeyMetadata{
+				Description: "rule 2 error key description",
+				Impact: utypes.Impact{
+					Name:   "impact_2",
+					Impact: 2,
+				},
+				Likelihood: 3,
+			},
+			HasReason: false,
+		},
+	}
+
+	ruleContent := types.RulesMap{
+		"rule_1": {
+			Summary:    "rule 1 summary",
+			Reason:     "This reason is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long. This summary is more than 255 characters long.",
+			Resolution: "rule 1 resolution",
+			MoreInfo:   "rule 1 more info",
+			ErrorKeys:  errorKeys,
+			HasReason:  true,
+		},
+		"rule_2": {
+			Summary:    "rule 2 summary",
+			Reason:     "rule 2 reason",
+			Resolution: "rule 2 resolution",
+			MoreInfo:   "rule 2 more info",
+			ErrorKeys:  errorKeys,
+			HasReason:  false,
+		},
+	}
+
+	cluster := types.ClusterEntry{
+		OrgID:         1,
+		AccountNumber: 1,
+		ClusterName:   "first_cluster",
+		KafkaOffset:   0,
+		UpdatedAt:     types.Timestamp(testTimestamp),
+	}
+
+	var deserialized types.Report
+	reportsJSON := types.ClusterReport("{\"analysis_metadata\":{\"metadata\":\"some metadata\"},\"reports\":[{\"rule_id\":\"rule_1|RULE_1\",\"component\":\"ccx_rules_ocp.external.rules.rule_1.report\",\"type\":\"rule\",\"key\":\"RULE_1\",\"details\":\"some details\"},{\"rule_id\":\"rule_2|RULE_2\",\"component\":\"ccx_rules_ocp.external.rules.rule_2.report\",\"type\":\"rule\",\"key\":\"RULE_2\",\"details\":\"some details\"}]}")
+	err := json.Unmarshal([]byte(reportsJSON), &deserialized)
+	if err != nil {
+		log.Fatal().Msg(err.Error())
+	}
+	reports := deserialized.Reports
+
+	producerMock := mocks.Producer{}
+	producerMock.On("ProduceMessage", mock.AnythingOfType("types.ProducerMessage")).Return(
+		func(msg types.ProducerMessage) int32 {
+			return 0
+		},
+		func(msg types.ProducerMessage) int64 {
+			return 0
+		},
+		func(msg types.ProducerMessage) error {
+			return nil
+		},
+	)
+
+	originalNotifier := serviceLogNotifier
+	serviceLogNotifier = &producerMock
+
+	produceEntriesToServiceLog(config, cluster, ruleContent, reports)
```

Comment:
no asserts after this call? since the notifier is a mock we can check if `ProduceMessage` has been called https://pkg.go.dev/github.com/stretchr/testify/mock#Mock.AssertNumberOfCalls . I'm not sure tho, you have to check

Reply:
You are right, I added assert :)